### PR TITLE
CE-1400 Adding WikiFactoryPublicStatusChanged hook

### PR DESCRIPTION
**Pull request by kamilkoterba**
https://wikia-inc.atlassian.net/browse/CE-1400

As a part of data consistency verification and consultation with @wladekb decided to change where we hook into wiki closing process. This way we can be sure to cover all closing cases like dead wikis automated task. 

Edit: Added skipping anon users here as well as it's data consistency thread already.

@Wikia/community-engineering

### CODE REVIEW

File: extensions/wikia/ExactTargetUpdates/hooks/ExactTargetUser.hooks.php [hunk 1]
```diff
@@ -25,7 +25,9 @@ public function onUserRenameAfterAccountRename( $iUserId, $sOldUsername, $sNewUs
 	 * @return bool
 	 */
 	public function onArticleSaveComplete( \WikiPage $article, \User $oUser ) {
-		$this->addTheUpdateCreateUserTask( $oUser );
+		if ( !$oUser->isAnon() ) {
+			$this->addTheUpdateCreateUserTask($oUser);
```
Comment (adamkarminski) [inline extensions/wikia/ExactTargetUpdates/hooks/ExactTargetUser.hooks.php]: Spaces per convention :)

File: extensions/wikia/WikiFactory/WikiFactory.php [hunk 2]
```diff
@@ -1849,6 +1849,8 @@ static public function setPublicStatus( $city_public, $city_id, $reason = "" ) {
 			__METHOD__
 		);
 
+		wfRunHooks( 'WikiFactoryPublicStatusChanged', array( &$city_public, &$city_id, $reason ) );
```
Comment (adamkarminski) [inline extensions/wikia/WikiFactory/WikiFactory.php]: Short array notation ```[]``` :)

Comment (nandy-andy) [inline extensions/wikia/WikiFactory/WikiFactory.php]: +1 :japanese_ogre:

**Comment by adamkarminski** [comment]
A few minor comments, when addressed - LGTM :+1:

File: extensions/wikia/ExactTargetUpdates/hooks/ExactTargetWiki.hooks.php [hunk 3]
```diff
@@ -55,12 +55,14 @@ public function onCityCatMappingUpdated( Array $aParams ) {
...
-	 * Executed on a WikiFactoryWikiClosed hook.
+	 * Executed on a WikiFactoryPublicStatusChanged hook.
 	 * @param  Array  $aParams Must contain a city_id key
 	 * @return true
 	 */
-	public function onWikiFactoryWikiClosed( Array $aParams ) {
-		$this->addTheDeleteWikiTask( $aParams );
+	public function onWikiFactoryPublicStatusChanged( &$city_public, &$city_id, $reason ) {
+		if ( $city_public <= 0 ) {
+			$this->addTheDeleteWikiTask( [ 'city_id' => $city_id ] );
```
Comment (wladekb) [inline extensions/wikia/ExactTargetUpdates/hooks/ExactTargetWiki.hooks.php]: Why do we need "The" in the name of the function?

File: extensions/wikia/WikiFactory/WikiFactory.php (hunk 2 above)
Comment (wladekb) [inline extensions/wikia/WikiFactory/WikiFactory.php]: There's no need to use references here. Please remove them.

Comment (kamilkoterba) [inline extensions/wikia/WikiFactory/WikiFactory.php]: Be careful what you ask for :japanese_ogre: 
Notation shortened.

File: extensions/wikia/ExactTargetUpdates/hooks/ExactTargetWiki.hooks.php (hunk 3 above)
Comment (kamilkoterba) [inline extensions/wikia/ExactTargetUpdates/hooks/ExactTargetWiki.hooks.php]: To be more clear that add is verb and delete is adjective ;)

**Pull request merged**